src/main/kotlin/ch/unibas/dmi/dbis/cottontail/database/column/mapdb/MapDBColumn.kt

```diff
@@ -691,7 +692,7 @@ internal class MapDBColumn<T : Any>(override val name: Name, override val parent
             if (this.status == TransactionStatus.CLOSED) throw TransactionException.TransactionClosedException(tid)
             if (this.status == TransactionStatus.ERROR) throw TransactionException.TransactionInErrorException(tid)
             if (this.status != TransactionStatus.DIRTY) {
-                if (this@MapDBColumn.txLock.writeLock().tryLock()) {
+                if (this@MapDBColumn.txLock.writeLock().tryLock(1, TimeUnit.SECONDS)) {
```

Comment:
Not sure, if this is the way to go. But I'll have a look.

Reply:
For highly concurrent transactions, waiting at most a second for other transactions to clean up  makes sense to me